### PR TITLE
Adds new buildable ammo printer - The Bluespace ammo manipulator

### PR DESCRIPTION
**Pull request by Majkl-J**
## About The Pull Request
Adds a new buildable ammo printer, requiring T4 parts to be constructed.
Locked under the same research that bluespace miners are.

closes: #81

### Progress

- [x] Add the new ammo printer
- [x] Figure out how I broke the ruin ammo printer
- [x] Add sprites to the Bluespace ammo manufacturer
- [x] Make it researchable
- [x] Realize that there was probably a easier way and refactor everything AGAIN
- [x] Realize that there was probably a easier way and refactor everything FOR THE THIRD FUCKING TIME

## Why It's Good For The Game
See #81 I was only hired to do the dirty bidding.

## Changelog
:cl:
add: Introduced the Bluespace ammo manufacturer, allowing crews to make ballistic ammo on the go.
/:cl:

### CODE REVIEW

**Review by Veiyna — CHANGES_REQUESTED** [review]
I've tested your PR and there are some things that could be improved

- The circuit and the design should be moved to the voidcrew modular directory
- The ruin ammo printer should probably be made a subtype of the normal ammo printer, and not the other way
- The code used to update the description for the metal amount is duplicated
- Description of the ruin ammo printer doesn't update correctly once it has been used up.
- all_nodes.dm has an unnecessary newline added

So far nothing else has caught my attention, but please make sure to test your changes in-game.

**Comment by Majkl-J** [comment]
> So far nothing else has caught my attention, but please make sure to test your changes in-game.

Oh all worked correctly the descriptions were refactored last minute and I didn't test them as in depth as I should've
Time to refactor the whole ruin stuff as subtype

File: voidcrew/code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -22,9 +22,9 @@
 
 /datum/techweb_node/bs_mining
 	id = "bluespace_mining"
-	display_name = "Bluespace Mining Technology"
-	description = "Harness the power of bluespace to make materials out of nothing. Slowly."
-	prereq_ids = list("bluespace_storage", "adv_mining")
+	display_name = "Bluespace Manufacturing Technology"
+	description = "Harness the power of bluespace to make materials out of nothing. And ammo."
+	prereq_ids = list("bluespace_storage", "adv_mining", "adv_weaponry")
 	design_ids = list("bluespace_miner", "miningcore_shell")
```
Comment (Veiyna) [inline voidcrew/code/modules/research/techweb/all_nodes.dm:28]: You'll want to add the ammo printer design here so it can be printed

Comment (Majkl-J) [reply voidcrew/code/modules/research/techweb/all_nodes.dm:28]: I appear to be an idiot

**Comment by Veiyna** [comment]
To the design_ids var, not prereq_ids

**Comment by Majkl-J** [comment]
> To the design_ids var, not prereq_ids

UUUUUUUUUUUUUUUGH I AM AN IDIOT TODAY

File: voidcrew/code/modules/research/designs/machine_designs.dm [hunk 2]
```diff
@@ -46,6 +46,14 @@
 	category = list("Misc. Machinery")
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE | DEPARTMENTAL_FLAG_CARGO | DEPARTMENTAL_FLAG_ENGINEERING
 
+/datum/design/board/ammo_printer
+	name = "Machine Design (Ammo Printer Board)"
+	desc = "The circuit board for an Ammo Printer."
+	id = "ammoprinter"
```
Comment (Veiyna) [inline voidcrew/code/modules/research/designs/machine_designs.dm:52]:
```suggestion
	id = "ammo_printer"
```

Comment (isotope436) [reply voidcrew/code/modules/research/designs/machine_designs.dm:52]: wont this break the research node if its changed?

Comment (Veiyna) [reply voidcrew/code/modules/research/designs/machine_designs.dm:52]: > wont this break the research node if its changed?

the techweb node design id is ammo_printer, so both of them have to match for the circuit to be printable in R&D

**Review by Veiyna — APPROVED** [review]
Everything seems to be fine now. 
I'll put it up on TM and if there aren't any issues it'll be merged.
We'll probably work on porting an ammo fabricator from another codebase soon but for now that'll do great for our needs!

**Pull request merged**